haystack/templates/predefined/indexing.yaml.jinja2

```diff
@@ -1,87 +1,113 @@
-    "init_parameters": {
-      "remove_empty_lines": true,
-      "remove_extra_whitespaces": true,
-      "remove_regex": None,
-      "remove_repeated_substrings": false,
-      "remove_substrings": None
-    },
-    "type": "haystack.components.preprocessors.document_cleaner.DocumentCleaner"
-  }, true) | tojson }}
-  embedder: {{ embedder | default({
-    "init_parameters": {
-      "batch_size": 32,
-      "device": {"type": "single", "device": "cpu"},
-      "embedding_separator": "\\n",
-      "model": "sentence-transformers/all-MiniLM-L6-v2",
-      "normalize_embeddings": false,
-      "prefix": "",
-      "progress_bar": true,
-      "suffix": "",
-      "token": None
-    },
-    "type": "haystack.components.embedders.sentence_transformers_document_embedder.SentenceTransformersDocumentEmbedder"
-  }, true) | tojson }}
+{% extends "base.yaml.jinja2" %}
+
+{% block components %}
+  cleaner:
+  {% if cleaner %}
+    {{ cleaner | indent }}
+  {% else %}
+    init_parameters:
+      remove_empty_lines: true
+      remove_extra_whitespaces: true
+      remove_regex: None
+      remove_repeated_substrings: false
+      remove_substrings: None
+    type: "haystack.components.preprocessors.document_cleaner.DocumentCleaner"
+  {% endif %}
+
+  embedder:
+  {% if embedder %}
+    {{ embedder | indent }}
+  {% else %}
+    init_parameters:
+      batch_size: 32
+      device:
+        type: "single"
+        device: "cpu"
+      embedding_separator: "\\n"
+      model: "sentence-transformers/all-MiniLM-L6-v2"
+      normalize_embeddings: false
+      prefix: ""
+      progress_bar: true
+      suffix: ""
+    type: "haystack.components.embedders.sentence_transformers_document_embedder.SentenceTransformersDocumentEmbedder"
+  {% endif %}
 
   # FileTypeRouter is used to route different file types to different file converters
   # The default mime types are set to text/plain. If we'll handle PDF files, we'll add application/pdf to mime types
   # Let's configure that part dynamically based on the use_pdf_file_converter flag
 
+  file_type_router:
+  {% with %}
   {% set default_mime_types = ["text/plain"] %}
   {% set additional_mime_types = ["application/pdf"] if use_pdf_file_converter | default(false) else [] %}
   {% set file_type_router_mime_types = default_mime_types + additional_mime_types %}
+  {% if file_type_router %}
+    {{ file_type_router | indent }}
+  {% else %}
+    init_parameters:
+      mime_types: {{ file_type_router_mime_types }}
+    type: "haystack.components.routers.file_type_router.FileTypeRouter"
+  {% endif %}
+  {% endwith %}
+
+  {% if doc_joiner %}
+    {{ doc_joiner | indent }}
+  {% else %}
+  doc_joiner:
+    init_parameters:
+      join_mode: "concatenate"
+    type: "haystack.components.joiners.document_joiner.DocumentJoiner"
+  {% endif %}
+
+  splitter:
+  {% if splitter %}
+    {{ splitter | indent }}
+  {% else %}
+    init_parameters:
+      split_by: "sentence"
+      split_length: 250
```

Comment:
This means 250 sentences per document, which is to long for processing with typical models. The model that is used here, all-MiniLM-L6-v2, has a token limit of 256 tokens. Let's use `split_by: "word"` instead and then we can keep `split_length: 250`.